### PR TITLE
Test for newlines at end of all *.rs and *.toml files

### LINKED ISSUE #160: Test for newline at the end of every .rs and .toml file
We want consistency here to avoid diffing problems later in git.

### PR DESCRIPTION
**Pull request by mbestavros**
Resolves #160.

### CODE REVIEW

**Comment by mbestavros** [comment]
I've rewritten the test to better follow the format we've established for our other tests. It also correctly tests all files now. Ready for review.

File: hooks/pre-push.d/eof-newline-test [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+top=`git rev-parse --show-toplevel`
+status=0
+for file in $(find . -path ./target -prune -o -type f -name "*.rs" -o -name "*.toml"); do
```
Comment (npmccallum) [inline hooks/pre-push.d/eof-newline-test]:
```suggestion
for file in $(find . -path ./target -prune -o -type f -name "*.rs" -o -type f -name "*.toml"); do
```

I think we need `-type f` again after the or.

File: hooks/pre-push.d/eof-newline-test [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    if [ -f "$file" ] && [ ! -z $(tail -c 1 "$file") ]; then
```
Comment (npmccallum) [inline hooks/pre-push.d/eof-newline-test]:
```suggestion
    if [ ! -z $(tail -c 1 "$file") ]; then
```

Also, this tests for any whitespace, not specifically newline.

File: hooks/pre-push.d/eof-newline-test [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+        status=1
+    fi
+done
+exit $status
+
+
+
+#  -print0 | xargs -0 -L1 bash -c 'test "$(tail -c 1 "$0")" && echo "$0"')
+# [ -z "$files_without_eof" ] && exit 0
+# printf "The following files are missing a newline at EOF:\n$files_without_eof\n" && exit 1
```
Comment (npmccallum) [inline hooks/pre-push.d/eof-newline-test]: Clean this up.

File: hooks/pre-push.d/eof-newline-test (hunk 1 above)
Comment (mbestavros) [reply hooks/pre-push.d/eof-newline-test]: Good catch. Fixed in latest push.

File: hooks/pre-push.d/eof-newline-test (hunk 2 above)
Comment (mbestavros) [reply hooks/pre-push.d/eof-newline-test]: Right. Fixed in latest push.

File: hooks/pre-push.d/eof-newline-test (hunk 3 above)
Comment (mbestavros) [reply hooks/pre-push.d/eof-newline-test]: D'oh, sorry. Fixed in latest push.

**Comment by mbestavros** [comment]
I've addressed the first round of feedback. I've also incorporated `-print` in the `find` invocation to suppress printing the `target` directory, as suggested by @npmccallum in #162.
